ddpui/tests/api_tests/test_airbyte_api.py

```diff
@@ -2,51 +2,1637 @@
+        airbyte_workspace_id="FAKE-WORKSPACE-ID", slug="test-org-slug"
+    )
+    yield org
+    print("deleting org_with_workspace")
+    org.delete()
+
+
+# ================================================================================
+def test_post_airbyte_detach_workspace_0():
+    """tests /worksspace/detatch/"""
+
+    mock_orguser = Mock()
+    mock_orguser.org = None
+
+    mock_request = Mock()
+    mock_request.orguser = mock_orguser
+
+    with pytest.raises(HttpError) as excinfo:
+        post_airbyte_detach_workspace(mock_request)
+    assert str(excinfo.value) == "create an organization first"
+
+
+def test_post_airbyte_detach_workspace_1(org_without_workspace):
+    """tests /worksspace/detatch/"""
+
+    mock_orguser = Mock()
+    mock_orguser.org = org_without_workspace
+
+    mock_request = Mock()
+    mock_request.orguser = mock_orguser
+
+    with pytest.raises(HttpError) as excinfo:
+        post_airbyte_detach_workspace(mock_request)
+    assert str(excinfo.value) == "org already has no workspace"
+
+
+def test_post_airbyte_detach_workspace_2(org_with_workspace):
+    """tests /worksspace/detatch/"""
+
+    mock_orguser = Mock()
+    mock_orguser.org = org_with_workspace
+
+    mock_request = Mock()
+    mock_request.orguser = mock_orguser
+
+    OrgPrefectBlock.objects.create(
+        org=mock_orguser.org,
+        block_type=ddpprefect.AIRBYTESERVER,
+        block_id="fake-serverblock-id",
+        block_name="fake-serverblock-name",
+    )
+    assert (
+        OrgPrefectBlock.objects.filter(
+            org=mock_orguser.org, block_type=ddpprefect.AIRBYTESERVER
+        ).count()
+        == 1
+    )
+    OrgPrefectBlock.objects.create(
+        org=mock_orguser.org,
+        block_type=ddpprefect.AIRBYTECONNECTION,
+        block_id="fake-connectionblock-id",
+        block_name="fake-connectionblock-name",
+    )
+    OrgPrefectBlock.objects.create(
+        org=mock_orguser.org,
+        block_type=ddpprefect.AIRBYTECONNECTION,
+        block_id="fake-connectionblock-id-1",
+        block_name="fake-connectionblock-name-1",
+    )
+    OrgPrefectBlock.objects.create(
+        org=mock_orguser.org,
+        block_type=ddpprefect.AIRBYTECONNECTION,
+        block_id="fake-connectionblock-id-2",
+        block_name="fake-connectionblock-name-2",
+    )
+    assert (
+        OrgPrefectBlock.objects.filter(
+            org=mock_orguser.org, block_type=ddpprefect.AIRBYTECONNECTION
+        ).count()
+        == 3
+    )
+
+    # @patch("ddpui.ddpprefect.prefect_service.delete_airbyte_server_block")
+    # def patched_delete_airbyte_server_block():
+    #     pass
+
+    @patch("ddpui.ddpprefect.prefect_service.delete_airbyte_connection_block")
+    def mock_delete_airbyte_connection_block():
+        pass
+
+    @patch("ddpui.ddpprefect.prefect_service.delete_airbyte_server_block")
+    def mock_delete_airbyte_server_block():
+        pass
+
+    post_airbyte_detach_workspace(mock_request)
```

Comment:
Looks like it is trying to connect to proxy server on calling `post_airbyte_detach_workspace`

`requests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=8080): Max retries exceeded with url: /delete-a-block/fake-serverblock-id`